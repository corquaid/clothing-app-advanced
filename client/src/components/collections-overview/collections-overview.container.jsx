import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { compose } from 'redux' // this import allows us to pass in all the HOCs by calling the function, USING CURRYING

import { selectIsCollectionFetching } from "../../redux/shop/shop.selectors";

import WithSpinner from "../with-spinner/with-spinner";
import CollectionsOverview from "../collections-overview/collections-overview";

const mapStateToProps = createStructuredSelector({
    isLoading: selectIsCollectionFetching,
}); // this will provide the isLoading prop to the WithSpinner component which is expecting the isLoading prop

const CollectionsOverviewContainer = compose(
    connect(mapStateToProps),
    WithSpinner
    )(CollectionsOverview);
    // this is using Currying to make more readable code and easier follow the sequence of what is happening

export default CollectionsOverviewContainer;
