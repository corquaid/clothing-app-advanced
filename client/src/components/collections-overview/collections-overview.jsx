import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectCollectionsForPreview } from '../../redux/shop/shop.selectors';
import CollectionPreview from '../collection-preview/collection-preview';

import './collections-overview.scss';

export const CollectionsOverview = ({ collections }) => (
  <div className='collections-overview'>
    {
    collections.map(({ id, ...otherCollectionProps}) => (
        <CollectionPreview className="collection-preview" key={id} {...otherCollectionProps} />
      ))
    }
  </div>
);

const mapStateToProps = createStructuredSelector({
  collections: selectCollectionsForPreview
})

export default connect(mapStateToProps)(CollectionsOverview);