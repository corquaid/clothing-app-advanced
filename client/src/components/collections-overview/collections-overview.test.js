import React from "react";
import { shallow } from "enzyme";
import { CollectionsOverview } from "./collections-overview";

// no grouping required

let wrapper;

const mockCollections = [
    {
        id: 1,
    },
    {
        id: 2,
    },
];

beforeEach(() => {
    wrapper = shallow(<CollectionsOverview collections={mockCollections} />);
});

it("should render the CollectionsOverview component", () => {
    expect(wrapper).toMatchSnapshot();
   
});

it('should render the same number of child components as the length of collections prop', () => {
    expect(wrapper.find('.collection-preview').length).toEqual(mockCollections.length);
})
