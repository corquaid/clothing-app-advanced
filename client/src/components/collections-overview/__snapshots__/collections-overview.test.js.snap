// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the CollectionsOverview component 1`] = `ShallowWrapper {}`;
